airflow/providers/amazon/aws/operators/glacier.py

```diff
@@ -0,0 +1,92 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.glacier import GlacierHook
+from airflow.utils.decorators import apply_defaults
+
+
+class GlacierCreateJobOperator(BaseOperator):
+    """
+    Initiate an Amazon Glacier inventory-retrieval job
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GlacierCreateJobOperator`
+
+    :param aws_conn_id: The reference to the AWS connection details
+    :type aws_conn_id: str
+    :param vault_name: the Glacier vault on which job is executed
+    :type vault_name: str
+    """
+
+    template_fields = ("vault_name",)
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        aws_conn_id="aws_default",
+        vault_name: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.aws_conn_id = aws_conn_id
+        self.vault_name = vault_name
+
+    def execute(self, context):
+        hook = GlacierHook(aws_conn_id=self.aws_conn_id)
+        response = hook.retrieve_inventory(vault_name=self.vault_name)
+        return response
+
+
+class GlacierDownloadArchiveOperator(BaseOperator):
+    """
+    This operator retrieves results from GlacierCreateJobOperator
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GlacierDownloadArchiveOperator`
+
+    :param aws_conn_id: The reference to the AWS connection details
+    :type aws_conn_id: str
+    :param vault_name: the Glacier vault on which job is executed
+    :type vault_name: string
+    :param job_id: the job ID was returned by retrieve_inventory()
+    :type job_id: str
+    """
+
+    template_fields = ("vault_name", "job_id")
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        aws_conn_id="aws_default",
+        vault_name: str,
+        job_id: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.aws_conn_id = aws_conn_id
+        self.vault_name = vault_name
+        self.job_id = job_id
+
+    def execute(self, context):
+        hook = GlacierHook(aws_conn_id="aws_default")
+        response = hook.retrieve_inventory_results(vault_name=self.vault_name, job_id=self.job_id)
+        return response
```

Comment:
I think this operator has not sense, it returns the following response (according to docs):
```
{
    'body': StreamingBody(),
    'checksum': 'string',
    'status': 123,
    'contentRange': 'string',
    'acceptRanges': 'string',
    'contentType': 'string',
    'archiveDescription': 'string'
}
```
If anything whould happend here is reading content of the file using the `StreamingBody` but where should it be saved?

Reply:
I agree.